core/query/sql.ml

```diff
@@ -137,138 +152,162 @@ let order_by_clause n =
    returned. This allows these operators to take lists that have any
    element type at all. *)
 
-let rec string_of_query quote ignore_fields q =
-  let sq = string_of_query quote ignore_fields in
-  let sb = string_of_base quote false in
-  let sbt = string_of_base quote true in
-  let string_of_fields fields =
+let pp_constant ppf s = Format.fprintf ppf "%s" s
+let pp_comma ppf () = Format.fprintf ppf ","
+
+let rec pr_query ppf quote ignore_fields q =
+  let pp_quote ppf q = Format.fprintf ppf "%s" (quote q) in
+
+  let pr_q ppf q = pr_query ppf quote ignore_fields q in
+  let pr_b ppf q = pr_base ppf quote false q in
+  let pr_b_ignore_fields ppf q = pr_base ppf quote true q in
+
+  let gen_pp_pair ppf fmt_str fl fr (l, r) = Format.fprintf ppf fmt_str fl l fr r in
+  let gen_pp_option ppf fmt_str f option =
+    match option with
```

Comment:
You could probably use `OptionUtils.opt_iter` here (defined in `Utililty`) rather than doing the split explicitly